dev/com.ibm.ws.sipcontainer/src/com/ibm/ws/sip/container/parser/ServletsInstanceHolder.java

```diff
@@ -42,70 +45,70 @@ public class ServletsInstanceHolder implements ServletInstanceHolderInterface{
      * Class Logger. 
      */
     private static final LogMgr c_logger = Log.get(ServletsInstanceHolder.class);
+    /* members */
+    private final static ServletsInstanceHolder s_instance = new ServletsInstanceHolder();
+    private SipContainer m_sipcontainer = SipContainer.getInstance();
 
-	/* members */
-	private final static ServletsInstanceHolder s_instance = new ServletsInstanceHolder();
-	private SipContainer m_sipcontainer = SipContainer.getInstance();
-	
-	private ThreadLocal<InitMembers> sipServletThreadLocal = new ThreadLocal<InitMembers>(); 
-	
-	private ServletsInstanceHolder() {
-		// Exists only to defeat instantiation.
-		if(c_logger.isTraceDebugEnabled()){
-			c_logger.traceDebug(this,"ServletsInstanceHolder","ServletsInstanceHolder constructor");	
-		}
-	}
-	public static ServletsInstanceHolder getInstance(){
-		return s_instance;
-	}
-	
-	private static class InitMembers {
-		public SipServlet sipServlet;
-		public ServletContext sipletContext;
-		public SipAppDesc appDesc;
-		
-		public InitMembers(SipAppDesc appDesc, SipServlet sipServlet, ServletContext sipletContext) {
-			this.appDesc = appDesc; 
-			this.sipServlet = sipServlet; 
-			this.sipletContext = sipletContext;
-		}
-	}
-	
-	/**
-	 * Map that holds the siplets instances, order by Application name (i.e <appname,Map >).
-	 * each inner Map is order by <siplet-class-name,siplet-instance>
-	 */
-	private Map<String, Map<String, Object>> m_sipAppsServlets = new HashMap<String, Map<String, Object>>(1);
-	
-	/**
-	 * add instance
-	 * @param appName
-	 * @param instance
-	 */
-	public void addSipletInstance(String appName,String className,Object servletInstance){
-		if(c_logger.isTraceDebugEnabled()){
-			c_logger.traceDebug(this,"addSipletInstance","appName["+appName+"] class["+className+"] instance["+servletInstance+"]");	
-		}
-		
-		SipAppDesc sipApp = m_sipcontainer.getSipApp(appName);
-		if(c_logger.isTraceDebugEnabled()){
-			c_logger.traceDebug(this,"addSipletInstance","found sipAppDesc ["+sipApp+"]");	
-		}
-		
-		Map<String, Object> app = m_sipAppsServlets.get(appName);
-		if(app == null){
-			app = new HashMap<String, Object>(1);
-			m_sipAppsServlets.put(appName,app);
-			
-	        // Setup the listeners in this stage in order at least
+    private final Map<String, InitMembers> sipServletThreadLocal = new ConcurrentHashMap<>();
```

Comment:
I would rename sipServletThreadLocal since it's no longer a thread local variable.